nip-nostr-connect.md

```diff
@@ -0,0 +1,130 @@
+NIP-Nostr-Connect
+======
+
+Nostr Connect allows Nostr apps to connect with remote signing devices
+------------------------
+
+`draft` `optional` `author:tiero` `author:giowe` `author:vforvalerio87`
+
+## Rationale
+
+Having to enter your Nostr private key on each website or random app _sucks_.
+
+
+## `TL;DR`
+
+
+**App** (*typically a web app, it generates a random ephemeral keypair*) and **Wallet** (*typically a mobile app, it holds the private key of the user that represents his Nostr account*) send to each other `kind:4` encrypted DMs, using a relay of choice. 
+
+App prompts the Wallet to do things such as fetching the public key or signing events.
+
+The `content` field must be a JSONRPC-ish **request** or **response**.
```

Comment:
It matches the JSON-RPC 2.0 Request and Reponse. I added `-ish` because the rest of the protocol  (batching, subscriptions etc...) in not relevant using a Nostr transport instead of TCP.

Would you remove the `JSONRPC-ish` definition entirely?

Reply:
Makes sense to me, would you also add a link to the jsonrpc 2.0 spec?

Reply:
You know the jsonrpc 2.0 spec considers payloads without `"jsonrpc": "2.0"` attributes invalid? Screw this shit.

Reply:
What's the consensus here? Do we keep `method` and `params` only?

`id` can be replaced by the tag `#e` maybe?